horovod/spark/common/util.py

```diff
@@ -539,6 +543,46 @@ def _train_val_split(df, validation):
     return train_df, val_df, validation_ratio
 
 
+_FILE_AVAILABILITY_WAIT_TIMEOUT_SECS = \
+    int(os.environ.get('FILE_AVAILABILITY_WAIT_TIMEOUT_SECS', '30'))
+
+
+def _wait_file_available(store, url_list):
+    """Waiting about _FILE_AVAILABILITY_WAIT_TIMEOUT_SECS seconds (default 30 seconds) to make sure
+    all files are available for reading. This is useful in some filesystems, such as S3 which only
+    providing eventually consistency.
+    """
+    # Import LocalStore here to avoid circular import
+    from horovod.spark.common.store import LocalStore
+    if isinstance(store, LocalStore):
+        return
+
+    def wait_for_file(path):
+        end_time = time.time() + _FILE_AVAILABILITY_WAIT_TIMEOUT_SECS
+        while time.time() < end_time:
+            if store.exists(path):
+                return True
+            time.sleep(0.1)
+        return False
+
+    pool = ThreadPool(min(len(url_list), 64))
+    try:
+        results = pool.map(wait_for_file, url_list)
+        failed_list = [url for url, result in zip(url_list, results) if not result]
+        if failed_list:
+            raise RuntimeError('Timeout while waiting for all parquet-store files to appear at urls {failed_list},'
```

Comment:
This method assumes that `url_list` refers to a parquet file's files. Not ideal. It works perfectly well for any kind of `url_list` that `store.exists` supports. Is it really needed to mention parquet here?

This method is private so I might tend to be OK with it.